fenicsprecice/fenicsprecice.py

```diff
@@ -273,9 +286,47 @@ def initialize(self, coupling_subdomain, mesh, function_space, write_function=No
             Recommended time step value from preCICE.
         """
 
-        coords = function_space.tabulate_dof_coordinates()
+        write_function_space, write_function = None, None
+        if type(write_object) is Function:
+            write_function_space = write_object.function_space()
+            write_function = write_object
+        elif type(write_object) is FunctionSpace:
+            write_function_space = write_object
+            write_function = None
+        elif write_object is None:
+            print("Participant {} is read-only participant".format(self._config.get_participant_name()))
+        else:
+            raise Exception("Given write object is neither of type dolfin.functions.function.Function or "
+                            "dolfin.functions.functionspace.FunctionSpace")
+
+        if read_function_space is None and write_function_space:
+            self._coupling_type = CouplingMode.UNIDIR_WRITE
+            assert(self._config.get_write_data_name())
+            print("Participant {} is write-only participant".format(self._config.get_participant_name()))
+        elif read_function_space and write_function_space is None:
+            self._coupling_type = CouplingMode.UNIDIR_READ
+            assert(self._config.get_read_data_name())
```

Comment:
I would suggest to put the user feedback from above (https://github.com/precice/fenics-adapter/pull/101/files#diff-6837aaf707ece32d403dde56a219c19c3f3ddf26c753a20a3ac4259a4f7fe4aaR297) to this place. Then the code is easier to read. In the section above you can simply:
```
elif write_object is None:
    pass
````
We will also avoid unnecessary output, if `read_function_space is None and write_function_space is None`. Currently, you will first `print(print("Participant {} is read-only participant".format(self._config.get_participant_name())))`, if `write_function_space is None` (https://github.com/precice/fenics-adapter/pull/101/files#diff-6837aaf707ece32d403dde56a219c19c3f3ddf26c753a20a3ac4259a4f7fe4aaR296-R297). However, a few lines later we enter the `else` and `raise Exception(...)` (https://github.com/precice/fenics-adapter/pull/101/files#diff-6837aaf707ece32d403dde56a219c19c3f3ddf26c753a20a3ac4259a4f7fe4aaR312-R314).